web/html/src/manager/proxy/container-config.tsx

```diff
@@ -0,0 +1,377 @@
+      })
+      .map((fieldName) => {
+        const field = document.getElementById(fieldName);
+        if (field !== null && field instanceof HTMLInputElement && field.files != null) {
+          const file = field.files[0];
+          return new Promise((resolve) => {
+            const reader = new FileReader();
+            reader.onload = (e) => {
+              if (e.target?.result instanceof ArrayBuffer) {
+                // Should never happen since we call readAsText, just quiets tsc
+                resolve(undefined);
+              } else {
+                resolve({ [fieldName]: e.target?.result });
+              }
+            };
+            reader.readAsText(file);
+          });
+        }
+        return undefined;
+      })
+      .filter((promise) => promise !== undefined);
+    Promise.all(fileReaders).then((values) => {
+      const commonData = {
+        proxyFQDN: model.proxyFQDN,
+        proxyPort: model.proxyPort ? parseInt(model.proxyPort, 10) : 8022,
+        serverFQDN: model.serverFQDN,
+        maxSquidCacheSize: parseInt(model.maxSquidCacheSize, 10),
+        proxyAdminEmail: model.proxyAdminEmail,
+        sslMode: model.sslMode,
+      };
+
+      const cnamesData = Object.fromEntries(Object.entries(model).filter(([key]) => key.startsWith("cnames")));
+      const extraData =
+        model.sslMode === SSLMode.CreateSSL
+          ? Object.assign(
+              {},
+              {
+                caPassword: model.caPassword,
+                country: model.country,
+                state: model.state,
+                city: model.city,
+                org: model.org,
+                orgUnit: model.orgUnit,
+                sslEmail: model.sslEmail,
+              },
+              cnamesData
+            )
+          : {};
+      const formData = unflattenModel(Object.assign({}, commonData, extraData, ...values));
+      Network.post("/rhn/manager/api/proxy/container-config", formData).then(
+        (data) => {
+          setSuccess(data.success);
+          setMessages([]);
+          setLoading(false);
+          window.location.assign("/rhn/manager/api/proxy/container-config/" + data);
+        },
+        (xhr) => {
+          try {
+            setSuccess(false);
+            setMessages([JSON.parse(xhr.responseText)]);
+            setLoading(false);
+          } catch (err) {
+            const errMessages =
+              xhr.status === 0
+                ? t("Request interrupted or invalid response received from the server.")
+                : Network.errorMessageByStatus(xhr.status)[0];
+            setSuccess(false);
+            setMessages([errMessages]);
+            setLoading(false);
+          }
+        }
+      );
+    });
+  };
+
+  const clearFields = () => {
+    setModel(initialModel);
+  };
+
+  const onValidate = (isValidated: boolean) => {
+    setIsValidated(isValidated);
+  };
+
+  const onChange = (newModel: any) => {
```

Comment:
Instead of `any`, you could do `Partial<typeof initialModel>` or extract that into a type. I'm also not opposed to just removing the type annotation here, but I think `any` is a bit misleading.

Reply:
Using `Partial<typeof initialModel>` gives a type error at the `setModel()` line, and no annotation gives a `better type may be inferred from usage`...